nats/aio/subscription.py

```diff
@@ -84,9 +84,9 @@ def __init__(
         self._pending_queue: asyncio.Queue[Msg] = asyncio.Queue(
             maxsize=pending_msgs_limit
         )
-        self._pending_size = 0
-        self._wait_for_msgs_task = None
-        self._message_iterator = None
+        self._pending_size: int = 0
+        self._wait_for_msgs_task: asyncio.Task | None = None
```

Comment:
Annotation with `|` are supported since Py3.8
Cuz the library still supports 3.7 for any time, this code should use Optional, Union, etc
Please, setup your formatter to check 3.7 compatibility. All of these constuction can be easely upgraded by [this tool](https://github.com/asottile/pyupgrade) one time, but for now we should follow the Py3.7 syntax

Reply:
The file has `from __future__ import annotations` import at the beginning which makes all type annotations lazy and allows us to use the annotation syntax from future Python versions. See #386 and #388.

Reply:
Well, it was a smth new 4 me, thx